lib/client/api_matrix.es6.js

```diff
@@ -75,124 +138,79 @@ foam.CLASS({
           for ease of expression in client UI components.`,
       args: [
         {
-          name: 'releaseKeys',
-          typeName: 'StringArray',
-          documentation: `Valid release keys used to filter
-              releaseWebInterfaceJunctionDAO.`,
-        },
-        {
-          name: 'options',
-          typeName: 'JSON',
-          documentation: `An optional argument are used to filter matrix:
-              searchKey: The result interface/API must contains this String
-                value, ignore case.
-              releaseOptions: An JSON of form {<releaseKey>: true, ...}.
-                When set, the returned matrix are filtered so that it includes
-                only APIs associated with a <releaseKey>.
-              numAvailable: An integer, or integer array. When set, only
-                APIs supported by exactly "numAvailable" releases are returned.
-                When "numAvailable" is an array, any integer in it is a
-                valid number supported releases.`,
+          name: 'metadata',
+          typeName: 'org.chromium.apis.web.MatrixMetadata',
         },
       ],
       returns: {
         typeName: 'JSON',
-        documentation: `JSON of the form:
+        documentation: `org.chromium.apis.web.MatrixMetadata.matrix. JSON:
             {interfaceName: {<apiName>: {<ReleaseKey>: true, ...}...}...}...}
             This structure simplifies for displaying a nested table.`,
       },
-      code: function(releaseKeys, options) {
-        const id = this.runInParallel ? 0 : ++this.matrixId_;
-        return this.verifyReleaseKeys_(releaseKeys).then(() => {
-          if (id && this.matrixId_ !== id) return null;
+      code: function(metadata) {
+        const id = metadata.id;
```

Comment:
Is it correct that this ID-passing stuff is mostly driven by sharing a single instance of this class instead of constructing one instance per view?

Reply:
That's correct. I don't think having separate instances would be much simpler though. Since the `ApiMatrix` and `ApiMatrixController` both live in a worker, creating multiple instances would turn id-passing into service object creation, registration, then binding.

Reply:
I see. Same controller, multiple `APIMatrix`es. Yeah, that would probably work.